src/main/scala/akka/stream/contrib/Accumulate.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+
+/**
+ * This companion defines a factory for [[Accumulate]] instances, see [[Accumulate.apply]].
+ */
+object Accumulate {
+
+  /**
+   * Factory for [[Accumulate]] instances.
+   *
+   * @param zero zero value for folding
+   * @param f binary operation for folding
+   * @tparam A input type
+   * @tparam B output type
+   * @return [[Accumulate]] instance
+   */
+  def apply[A, B](zero: B)(f: (B, A) => B): Accumulate[A, B] = new Accumulate(zero)(f)
```

Comment:
we are not on 2.12 yet, are you?

I think we introduced akka.japi.function.* because of variance (and akka streams were originally compiled with java6)